### ISSUE TITLE
How to `bundle` when upgrading from version 7 to 12

### ISSUE THREAD

**Issue by davidseek** [react-native-community/cli#2275]
I'm in the process of updating an app from React-native 0.68.1 to 0.73.2

We're using [react-native/local-cli](https://github.com/facebook/react-native/blob/v0.73.2/packages/react-native/local-cli/cli.js#L12) to run:

```
executing "./node_modules/react-native/local-cli/cli.js bundle --dev true --entry-file ....
```

With 68, we where using [react-native-community/cli@7](https://github.com/facebook/react-native/blob/v0.68.1/package.json#L90) as dependency of `react-native`. 

With 73, the version has changed [react-native-community/cli@12](https://github.com/facebook/react-native/blob/v0.73.2/packages/react-native/package.json#L96).

Version [7 had metroCommands](https://github.com/react-native-community/cli/blob/v7.0.1/packages/cli/src/commands/index.ts#L14). Version [12 does not](https://github.com/react-native-community/cli/blob/v12.3.3/packages/cli/src/commands/index.ts#L9). I assume, that the `bundle` command came from `metroCommands`.

Now, just as expected I'm getting `error: unknown command 'bundle'`, because the bundle command doesn't exist anymore.

`npx react-native bundle` causes the same issue.

Since it resolves to the same version, the bundle command is missing, too.

```
npx react-native -v  
12.3.0

npx react-native bundle
error: unknown command 'bundle'
```

How can I use the `bundle` command going forward? Any pointers are highly appreciated.

**Comment by davidseek** [comment]
PS, I found [@react-native/community-cli-plugin](https://github.com/facebook/react-native/blob/v0.73.2/packages/community-cli-plugin/README.md), which seems to carry the `bundle` command and is marked as "Formerly [@react-native-community/cli-plugin-metro](https://www.npmjs.com/package/@react-native-community/cli-plugin-metro)".

But the [usage](https://github.com/facebook/react-native/blob/v0.73.2/packages/community-cli-plugin/README.md#usage-1) leads me back to my original issue: 

```
npx react-native bundle --entry-file <path> [options]
```

**Comment by szymonrybczak** [comment]
hey @davidseek! that's strange these commands should be added by Core thanks to `@react-native/community-cli-plugin`, please make sure that you have proper `react-native` version, and also do you mind running `npx react-native config`?

**Comment by davidseek** [comment]
Thanks for getting back to me. The problem seems to be the following:

My workspace has 2 npm packages.
- Package A contains all node_modules, aka 3p dependencies
- Package B is my Manifest, it syncs the node_modules from Package A

Running bundle command from Package A works.
```
Package A -> node node_modules/react-native/local-cli/cli.js bundle
                Welcome to Metro v0.80.4
              Fast - Scalable - Integrated
```

Running bundle command from Package B does not work.
```
Package B -> node node_modules/react-native/local-cli/cli.js bundle
error: unknown command 'bundle'
```

Even when I try to call the script from Package A from within Package B it also throws the same issue. 
```
Package B -> node ../Package A/node_modules/react-native/local-cli/cli.js bundle 
error: unknown command 'bundle'
```

Only works if I call it directly from the root of Package A.
```
Package A -> node node_modules/react-native/local-cli/cli.js bundle
                Welcome to Metro v0.80.4
              Fast - Scalable - Integrated
```

`node_modules` of both packages are in sync. I understand that you're not exactly supporting custom setups, but any clue why Package A can find the `bundle` info from within `@react-native`, as you described, but Package B can't?

**Comment by szymonrybczak** [comment]
1. Does running via `npx react-native` change something? 
2. Could you please confirm if the config return from `npx react-native config` command from package A is the same as from package B?

**Comment by davidseek** [comment]
- `npx react-native` works in Package A, just like calling `cli.js bundle` directly
- `npx react-native` in Package B leads to `unknown command 'bundle'`

Package B `npx react-native config` output: 
```
{
  "root": "/Volumes/workplace/Package B",
  "reactNativePath": "/Volumes/workplace/Package B/node_modules/react-native",
  "reactNativeVersion": "0.73",
  "dependencies": {},
  "commands": [],
  "healthChecks": [],
  "platforms": {},
  "project": {}
}
```

Package A `npx react-native config` output shows a WHOLE LOT MORE, and also contains the bundle command.

```
"root": "/Volumes/workplace/Package A",
"reactNativePath": "/Volumes/workplace/Package A",
"reactNativeVersion": "0.73",
....
"commands": [
{ "name": "bundle",
      "description": "Build the bundle for the provided JavaScript entry file.",
      "options": [
        {
          "name": "--entry-file <path>",
          "description": "Path to the root JS file, either absolute or relative to JS root"
        },
        {
        
...
            
"healthChecks": [],
"platforms": {
    "ios": {},
    "android": {}
  },
"project": {
    "ios": null,
    "android": null
  }
  ```

How can I sync this info from Package A to Package B?

**Comment by SemihGk** [comment]
I have the same issue `:error: unknown command 'bundle' `. I am also on the process of upgrading from 0.69 to 0.73. After debugging `@react-native-community/cli `and `@react-native-community/cli-config` libraries, It looks like your metro.config.js file should be under same level of` node_modules/@react-native-community/cli**` folders

Have the following monorepo folder structure:

```
apps/
     - mobile/
        -- metro.config.js
        -- react-native.js
node_modules/
      - @react-native-community/
         -- cli/
         -- cli-config/   

```
Eventually,I had to create symlinks to the config files folder in order to resolve the issue.

```
ln -snf node_modules/@react-native-community apps/mobile/node_modules
ln -snf node_modules/react-native apps/mobile/node_modules
```

Obviously, this cannot be a desired solution. Hopefully, we can find a better solution. It is an annoying issue indeed.

**Comment by davidseek** [comment]
We have the following structure

```
packageA/
    - node_nodules (original)
packageB/
    - node_nodules (symlink of packageA/node_nodules)
    - script for bundle creation
    - metro config etc
```

We used to call the `script for bundle creation` from packageB root. This throws now `unknown command 'bundle'`. I'm able to call the same underlying command, equivalent to `npx react-native bundle ...` from packageA and it works. But all my Configs and sources are in packageB. Metro config missing etc.

Symlinking the `node_nodules` content as you did above @SemihGk wouldn't solve my issue. Because we're already doing that. I need to be able to call `npx react-native bundle ...` from packageB.

**Comment by davidseek** [comment]
I found a way to bypass my issue. The key is, that `packageB` has no direct dependencies. `packageA` is the 3p dependencies repo which other packages symlink from. 

In order to use the `bundle` command, we had to add `react-native` as a dependency within `packageB`.

Without running `yarn install` or `npm install`, `npx react-native bundle` works now as expected.

For us the key is not to actually have direct dependencies outside of `packageA`. 

Therefore I'm using a small script to add the dependency right before we build the `bundle`.

```
// Pseudo
const json = await fs.readFile('package.json)
json.dependencies = { "react-native": "^0.73.2" }
await fs.writeFile('package.json, json)
```
